src/core/env.js

```diff
@@ -113,12 +113,12 @@ const defaults = new Map(
       type: "string",
       default: "https://cfstore.rethinkdns.com/blocklists/",
     },
-    // primary doh upstream
+    // primary upstream (DoH, DoT, DoQ upstreams are compatible)
```

Comment:
*workers* cannot connect to DoT or DoQ upstreams. `fetch`, which is a "WebAPI", is HTTP only.

Reply:
So if you specify a DoQ upstream such as the one in my comment above, does it reroute to a DoH upstream?

Reply:
I'm not even sure if it works. But if it does, then it is most likely doing DoH, instead.

Reply:
It somehow works for me through both controld and nextdns in my testing. If you are aware of any way to capture what protocol is actually being sent upstream from the providers, then I could check to see if it is doing DoH which it likely is

Reply:
type `quic://` or `tls://` for an upstream that doesn't support those protocols (like `sky.rethinkdns.com`). If the upstream still resolves, then you know that `tls` and `quic` protocol schemes have no effect and that everything defaults to DoH on *Workers*.

Reply:
I know I already closed this PR but I have found the limits to when the service starts acting up with dns requests from quic/tls. If I specify port 853 by :853 at the end, it does not like to connect to any server. So yeah, it is rerouting through doh. 

Anyways, sorry for all my stupid pr requests I’m opening lately.

Reply:
Appreciate it. You're the sole developer on serverless-dns right now as I recover from my ailments. (: So appreciate no matter what.

Reply:
> Btw, raw TCP (hence TLS) support will come to Workers (both incoming and outgoing TCP): https://blog.cloudflare.com/introducing-socket-workers/

Was it added [here](https://github.com/cloudflare/workerd/pull/162)

Reply:
Yes. That's for *ingress* over TCP (available in private beta, afaik).

Reply:
@ignoramous it appears tcp workers is out of private beta and is now apart of the public beta as of 3 days ago!

https://developers.cloudflare.com/workers/platform/betas/

Reply:
*Ingress* not supported, only *egress*.

Reply:
That’s a shame, well hopefully cloudflare will support ingress over TCP soon🤞